### PR TITLE
Added skani modules

### PR DESCRIPTION
**Pull request by CarsonJM**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/skani/dist/main.nf [hunk 1]
```diff
@@ -0,0 +1,50 @@
+process SKANI_DIST {
+    tag "$meta.id"
+    label 'process_high'
```
Comment (jfy133) [inline modules/nf-core/skani/dist/main.nf:3]: Seems a bit extreme!

```


Efficient database search. Querying a genome against a preprocessed database of >65000 prokaryotic genomes takes a few seconds with a single processor and ~6 GB of RAM. Constructing a database from genome sequences takes a few minutes to an hour.
```
```suggestion
    label 'process_medium'
```
Maybe that's more reasonable

File: modules/nf-core/skani/dist/main.nf [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    def args = task.ext.args ?: ''
+    prefix = task.ext.prefix ?: "${meta.id}_${meta2.id}"
+    """
+    skani \\
+        dist \\
+            -q ${query} \\
+            -r ${reference} \\
+            -o ${prefix}.tsv \\
+            -t ${task.cpus} \\
+            ${args}
```
Comment (jfy133) [inline modules/nf-core/skani/dist/main.nf:31]: I saw it has `-v` and `-t` do either include useful stats [^1] for potentai l MultiqC? stuff

[^1]: https://nf-co.re/docs/contributing/nfcore_component-specifications/modules#110-capturing-stdout-and-stderr

File: modules/nf-core/skani/dist/main.nf [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    prefix = task.ext.prefix ?: "${meta.id}_${meta2.id}"
+    """
+    skani \\
+        dist \\
+            -q ${query} \\
+            -r ${reference} \\
```
Comment (jfy133) [inline modules/nf-core/skani/dist/main.nf:28]: Should the `-rl`/`-ql` options be supported? |

```
        --ql <query list file>        File with each line containing one fasta/sketch file.
        --rl <reference list file>    File with each line containing one fasta/sketch file.

```

Might help reduce the number of parallel very small jobs if you were to do it one by one

File: modules/nf-core/skani/dist/meta.yml [hunk 4]
```diff
@@ -0,0 +1,56 @@
+name: "skani_dist"
+description: Simple ANI calculation between reference and query genomes.
+keywords:
+  - skani
```
Comment (jfy133) [inline modules/nf-core/skani/dist/meta.yml:4]:
```suggestion
  - ani
  - skani
```

File: modules/nf-core/skani/search/main.nf [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    tag "$meta.id"
+    label 'process_high'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/skani:0.2.1--h4ac6f70_0':
+        'biocontainers/skani:0.2.1--h4ac6f70_0' }"
+
+    input:
+    tuple val(meta) , path(query)
```
Comment (jfy133) [inline modules/nf-core/skani/search/main.nf:11]: Missing `-ql <query_list_file`

File: modules/nf-core/skani/search/meta.yml [hunk 6]
```diff
@@ -0,0 +1,54 @@
+name: "skani_search"
+description: Memory-efficient ANI database queries with skani.
+keywords:
+  - skani
```
Comment (jfy133) [inline modules/nf-core/skani/search/meta.yml:4]:
```suggestion
  - ani
  - skani
```
and in all other modules

File: modules/nf-core/skani/search/meta.yml [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. `[ id:'sample1' ]`
+  - search:
+      type: file
+      description: |
+        TSV file containing ANI and AF as predicted by skani.
+      pattern: "*.sketch"
```
Comment (jfy133) [inline modules/nf-core/skani/search/meta.yml:45]:
```suggestion
      pattern: "*.tsv"
```

No?

File: modules/nf-core/skani/sketch/main.nf [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    tag "$meta.id"
+    label 'process_high'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/skani:0.2.1--h4ac6f70_0':
+        'biocontainers/skani:0.2.1--h4ac6f70_0' }"
+
+    input:
+    tuple val(meta), path(fasta)
```
Comment (jfy133) [inline modules/nf-core/skani/sketch/main.nf:11]: `-l <fasta_list>` option?

**Review by jfy133 — COMMENTED** [review]
Note that the `-ql` etc. options may not be relevant, except maybe when you have millions of genomes and exceed UNIX amximum command lengths. But I've not used SKANI so I dunno how often that could be a problem.

File: modules/nf-core/skani/sketch/meta.yml [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+      doi: "10.1038/s41592-023-02018-3"
+      licence: ["MIT"]
+
+input:
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. `[ id:'sample1' ]`
+  - query:
```
Comment (jfy133) [inline modules/nf-core/skani/sketch/meta.yml:24]: It's called `fasta` in the module?

File: modules/nf-core/skani/triangle/main.nf [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+    tag "$meta.id"
+    label 'process_high'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/skani:0.2.1--h4ac6f70_0':
+        'biocontainers/skani:0.2.1--h4ac6f70_0' }"
+
+    input:
+    tuple val(meta) , path(queries)
```
Comment (jfy133) [inline modules/nf-core/skani/triangle/main.nf:11]: `-l` as before, 

quries are FASTA files right? Might be clearer to just call it that.

**Review by jfy133 — COMMENTED** [review]
Overall really good! Just some systematic suggestions (if --trace/--debug contain stats useful for possible multiqc, and supporting teh `list` options etc.)

File: modules/nf-core/skani/dist/main.nf (hunk 2 above)
Comment (CarsonJM) [reply modules/nf-core/skani/dist/main.nf:31]: Just checked and tbh the output is kind of meaningless to me...
`[00:00:00.000] (7f54a622e800) INFO   skani dist -q genome.fasta.gz -r contigs.fasta -o test_test2.tsv -t 2 --debug
[00:00:00.000] (7f54a622e800) DEBUG  Using C125 regression model.
[00:00:00.001] (7f54a622e800) INFO   Learned ANI mode detected. ANI may be adjusted according to a regression model trained on MAGs.
[00:00:00.001] (7f54a622e800) INFO   Generating sketch time: 0.00141721
[00:00:00.001] (7f54a622e800) DEBUG  Ref seeds len 230, Query seeds len 271, Anchors 228, Seeds hit query 224, Est 0.9873820971742308, Ref_file contigs.fasta, Query_file genome.fasta.gz
[00:00:00.001] (7f54a622e800) DEBUG  Query genome.fasta.gz Ref contigs.fasta - ANI 0.9993965571153403, +/- = 0/1. 
[00:00:00.001] (7f54a622e800) INFO   ANI calculation time: 0.00033303`

**Pull request closed without merging**